KeyboardKit/KeyboardInput.swift

```diff
@@ -0,0 +1,79 @@
+// Douglas Hill, December 2022
+
+/// Input required on a hardware keyboard (a combination of keys).
+///
+/// A keyboard command consists of input (this type) + action (code to run).
+struct KeyboardInput {
+    let modifierFlags: UIKeyModifierFlags
+    let character: String
+    let allowsAutomaticMirroring: Bool
+
+    // The default for mirroring doesn’t matter for most inputs since they’re not directional, but might as well match the UIKit and SwiftUI default.
+
+    init(_ modifierFlags: UIKeyModifierFlags, _ character: String, allowsAutomaticMirroring: Bool = true) {
+        self.modifierFlags = modifierFlags
+        self.character = character
+        self.allowsAutomaticMirroring = allowsAutomaticMirroring
+    }
+
+    /// Shorthand for when there are no modifier keys.
+    init(_ character: String, allowsAutomaticMirroring: Bool = true) {
+        self.init([], character, allowsAutomaticMirroring: allowsAutomaticMirroring)
+    }
```

Comment:
Should be able to set a default value for `modifierFlags` on the other initialise and remove this one.

Reply:
Actually I’m just going to delete this.